### DISCUSSION TITLE
Gotenberg docker image missing for armhf on Raspberry docker container (since update?)

### DISCUSSION THREAD

**Discussion by Roeschi85** [jonaswinkler/paperless-ng#1256] [Support]
Hi,
I'm running paperless-ng for several months on my Raspberry 4B now, but I had to roll out the paperless containers from scratch. I've been using docker-compose.

When I'm pulling all images with `docker-compose pull` everything but gotenberg gets pulled as it should. Gotenberg seems to have a new update and it says "ERROR: for gotenberg  no matching manifest for linux/arm/v7 in the manifest list entries.".

It looks like the latest image for gotenberg does not support armhf/armv7 until now. Where can I find a working image compatible with paperless-ng?

**Comment by Tooa** [comment]
Did the official Gotenberg docker image work up until recently on your RaspberryPi? I checked their [Dockerhub](https://hub.docker.com/r/thecodingmachine/gotenberg/tags) and couldn't verify armv7 or arm64 support for any of their older Gotenberg v6.x versions.

Haven't found another Gotenberg docker image suitable for RaspberryPi with armv7, when doing a quick google search either. You can raise a feature request over at the Gotenberg project and ask for support. But I don't think they will create an armv7 image, because Gotenberg v7.x introduced arm64 support. I think, you can utilize the latter with a 64-bit variant of RaspberryPi OS. The last time I checked though [their 64-bit variant ](https://downloads.raspberrypi.org/raspios_arm64/images/)where still in beta. 

It seems there is also no official support for armv7 or arm64 over at [Tika](https://hub.docker.com/r/apache/tika). So, you got another problem. I wonder how Tika worked for you in the past then?

Besides the missing docker images for RaspberryPi, Tika is quite resource hungry and might not run nice on your RaspberryPi. Cannot say for sure, because I don't use it on my RaspberryPi.

Hope that helps.

**Comment by Roeschi85** [comment]
Gotenberg was pulled on my last container rollout in august by docker-compose or the install script. You're right, there is no armv7 support mentioned on dockerhub (anymore?). I've tried it on a fresh Raspberry Pi OS 32bit, so it's just no docker image available. I was wondering if my system was broken... 

Tika image was pulled succesfully, beside the fact that there is no armv7 image on dockerhub. See #1257. Perfomance-wise paperless-ng with tika on a Raspberry 4 is not that bad and it would be very annoying, if paperless-ng loses support for Raspberries because of its dependencies to other images.

**Reply by iwishiwasaneagle** [reply]
I haven't been able to get Tika to run on RPi. What's your setup?

**Reply by TobenderZephyr** [reply]
Well, actually i found that the manifest includes arm/v7 for gotenberg:
https://hub.docker.com/r/gotenberg/gotenberg/tags?page=1&ordering=last_updated

But you're right, apache/tika only has amd64 versions.

What is your configuration in both `image:`-statements? Did you maybe build it yourself before (`build: .`)?

**Reply by iwishiwasaneagle** [reply]
I kinda just gave up on it but will try it again!

**Reply by iwishiwasaneagle** [reply]
Okay so I forked [apache/tika-docker](https://github.com/apache/tika-docker) to [iwishiwasaneagle/tika-docker](https://github.com/iwishiwasaneagle/tika-docker) and added a github actions workflow to build tika for arm64 and amd64. The actual gotenberg docker image worked flawlessly. I'm going to make a PR for this repo to create a new docker-compose with tika and gotenberg for RPi

**Reply by abhilesh** [reply]
I was able to successfully build `armhf` and `arm64` docker images for apache tika. You can find them at - https://hub.docker.com/r/abhilesh7/apache-tika-arm

I have a template docker-compose.yml for running Paperless-ng on Raspberry Pi 4 over at - https://github.com/abhilesh/self-hosted_docker_setups/tree/main/paperless-ng

I'd be happy to send in a PR if there is interest for this

**Reply by 0xSh1v4** [reply]
Hi, were you able to run the gotenburg on pi. I have only v7 available to arm and it has some issues. There is a another issue created here. https://github.com/jonaswinkler/paperless-ng/discussions/1256
I could not able to run gotenburg on mine either.

**Reply by abhilesh** [reply]
What issues are you facing? I don't have an `armhf` system for testing right now but I've been using the [thecodingmachine/gotenberg](https://hub.docker.com/r/thecodingmachine/gotenberg) docker image without any issues for a while now

**Reply by 0xSh1v4** [reply]
This is the issue : Error while converting document to PDF: 404 Client Error: Not Found for url: http://gotenberg:3000/convert/office
This thread mentions it https://github.com/jonaswinkler/paperless-ng/issues/1250

**Reply by abhilesh** [reply]
Is version 6 giving the same issues too? I'm using the `latest` tag from `thecodingmachine/gothenberg` and don't have those errors.

This [PR](https://github.com/jonaswinkler/paperless-ng/pull/1262) seems to have fixed that issue (see point 3)

**Reply by 0xSh1v4** [reply]
Yeah I saw that too. where can I change the endpoint url. I am running inside the docker container. Do you where can I modify this _Change endpoint from /convert/office to /forms/libreoffice/convert_?

**Reply by abhilesh** [reply]
That endpoint is located in `src/paperless_tika/parsers.py` but I see that the `master` branch is still using */convert/office* while the `dev` branch contains the changed endpoint and points to */forms/libreoffice/convert*

Till the change makes it over to the master branch, you can build the docker image locally by pointing your docker-compose file to the dev branch; make these modifications in your docker-compose file (omitting the irrelevant parts) - 
```yaml
services:
  webserver:
    build: https://github.com/jonaswinkler/paperless-ng.git#dev
```
This will instruct docker-compose to build the image locally using the `dev` branch

**Reply by 0xSh1v4** [reply]
Thanks. I just pulled the dev image and everything seems working fine. I will wait for the official release to use then.
